e2e/tests/migration_steps_test.go

```diff
@@ -90,45 +219,381 @@ func (s *migrationSuite) TheShannonServiceConfigIsUpdatedIfApplicable(actorType
 }
+
+func (s *migrationSuite) TheMorseClaimableAccountIsMarkedAsClaimedByTheShannonAccountAtARecentBlockHeight() {
+	var isShannonKeyFound bool
+	expectedMorseClaimableAccount := *s.morseClaimableAccount
+	expectedMorseClaimableAccount.ClaimedAtHeight = s.morseAccountClaimHeight
+	expectedMorseClaimableAccount.ShannonDestAddress, isShannonKeyFound = s.getShannonKeyAddress()
+	require.True(s, isShannonKeyFound)
+
+	*s.morseClaimableAccount = s.queryShowMorseClaimableAccount(s.morseClaimableAccount.MorseSrcAddress)
+	require.Equal(s, &expectedMorseClaimableAccount, s.morseClaimableAccount)
+}
+
+// runCommand executes the given command string and returns the output and any error.
+func (s *migrationSuite) runCommand(commandStr string) ([]byte, error) {
+	commandStringParts := strings.Split(commandStr, " ")
+
+	cmd := exec.Command(commandStringParts[0], commandStringParts[1:]...)
+	output, err := cmd.CombinedOutput()
+
+	return output, err
+}
+
+// writeTempFile creates a temporary file with the given fileName and content.
+// It returns the path to the temporary file and the temporary file is removed
+// when the test completes.
+func (s *migrationSuite) writeTempFile(fileName string, content []byte) string {
+	outputPath, err := os.CreateTemp("", fileName)
+	require.NoError(s, err)
+	defer func() {
+		_ = outputPath.Close()
+	}()
+
+	// Delete the temp file when the test completes.
+	s.Cleanup(func() {
+		_ = os.Remove(outputPath.Name())
+	})
+
+	_, err = outputPath.Write(content)
+	require.NoError(s, err)
+
+	return outputPath.Name()
+}
+
+// nextMorseKeyIdx increments the morseKeyIdx global variable and returns the
+// incremented value.
+func (s *migrationSuite) nextMorseKeyIdx() uint64 {
+	morseKeyIdx++
+	return morseKeyIdx
+}
+
+// getMorseKeyIdx returns the current value of the morseKeyIdx global variable.
+func (s *migrationSuite) getMorseKeyIdx() uint64 {
+	return morseKeyIdx
+}
+
+// nextMorseUnstakedKeyIdx returns the next morse private key index which is
+// intended to be used for unstaked morse accounts. If the current morseKeyIdx
+// is not an unstaked morse account, morseKeyIdx is incremented until the next
+// Morse key index which should be an unstaked account, given the round-robin
+// distribution of morse account actor types.
+func (s *migrationSuite) nextMorseUnstakedKeyIdx() uint64 {
+	currentIdx := s.getMorseKeyIdx()
+	// Skip non-application account keys.
+	for {
+		if testmigration.GetRoundRobinMorseAccountActorType(currentIdx) ==
+			testmigration.MorseUnstakedActor {
+			break
+		}
+		currentIdx = s.nextMorseKeyIdx()
+	}
+
+	return currentIdx
+}
+
+// nextShannonKeyIdx randomizes the shannon key index and returns a key name
+// which is derived from the new index.
+func (s *migrationSuite) nextShannonKeyIdx() string {
+	shannonKeyIdx = rand.Uint64()
+	return s.getShannonKeyName()
+}
+
+// getShannonKeyName returns the key name derived the current shannon key index.
```

Comment:
I understand the need for a global variable but I start getting confused by where we're being functional and where we're not.

What if we:
1. Keep `shannonKeyIdx` global
2. Have a helper called `getShannonKeyIdx` and `nextShannonKeyIdx`
3. Everything else is "functional" by using it as input params?

Reply:
The global variables are ONLY referenced by their respective `get...` and `next...` functions. Is this not already what you describe?